openmdao/core/component.py

```diff
@@ -615,12 +586,110 @@ def declare_partials(self, of, wrt, dependent=True,
+
+        # Approximation of the derivative, former API call approx_partials.
+        else:
+
+            if method not in self._approx_schemes:
+                self._approx_schemes[method] = supported_methods[method]()
+
+            # If rows/cols is specified
+            if rows is not None or cols is not None:
+                raise ValueError('Sparse FD specification not supported yet.')
+
+            # Need to declare the Jacobian element too.
+            self._declared_partials.append((of, wrt, True, rows, cols, val))
+
+            self._approximated_partials.append((of, wrt, method, kwargs))
+
+    def set_check_partial_options(self, wrt, method='fd', form=None, step=None, step_calc=None):
+        """
+        Set options that will be used for checking partial derivatives.
+
+        Parameters
+        ----------
+        wrt : str or list of str
+            The name or names of the variables that derivatives are taken with respect to.
+            This can contain the name of any input or output variable.
+            May also contain a glob pattern.
+        method : str
+            Method for check: "fd" for finite difference, "cs" for complex step.
+        form : str
+            Finite difference form for check, can be "forward", "central", or "backward". Leave
+            undeclared to keep unchanged from previous or default value.
+        step : float
+            Step size for finite difference check. Leave undeclared to keep unchanged from previous
+            or default value.
+        step_calc : str
+            Type of step calculation for check, can be "abs" for absolute (default) or "rel" for
+            relative.  Leave undeclared to keep unchanged from previous or default value.
+        """
+        supported_methods = {'fd': FiniteDifference,
+                             'cs': ComplexStep}
+
+        if method not in supported_methods:
+            msg = 'Method "{}" is not supported, method must be one of {}'
+            raise ValueError(msg.format(method, supported_methods.keys()))
+
+        wrt_list = [wrt] if isinstance(wrt, string_types) else wrt
+        self._declared_partial_checks.append((wrt_list, method, form, step, step_calc))
+
+    def _get_check_partial_options(self):
+        """
+        Return dictionary of partial options with pattern matches processed.
+
+        This is called by check_partials.
+
+        Returns
+        -------
+        dict(wrt : (options))
+            Dictionary keyed by name with tuples of options (method, form, step, step_calc)
+        """
+        opts = {}
+        outs = list(self._var_allprocs_prom2abs_list['output'].keys())
+        ins = list(self._var_allprocs_prom2abs_list['input'].keys())
+        for wrt_list, method, form, step, step_calc in self._declared_partial_checks:
+            for pattern in wrt_list:
+                wrt_matches = set(find_matches(pattern, outs + ins))
```

Comment:
no need to create a set here since you're only using it for iteration an not lookups.

Reply:
fixed